lib/matplotlib/backend_bases.py

```diff
@@ -3273,19 +3330,86 @@ def press_zoom(self, event):
         if (event.button not in [MouseButton.LEFT, MouseButton.RIGHT]
                 or event.x is None or event.y is None):
             return
-        axes = [a for a in self.canvas.figure.get_axes()
-                if a.in_axes(event) and a.get_navigate() and a.can_zoom()]
-        if not axes:
+
+        # Get all axes and group them by their navigation-capture behaviour.
+        axes_capture = {True: {}, False: {}}
+        for a in self.canvas.figure.get_axes():
+            if a.in_axes(event) and a.get_navigate() and a.can_zoom():
+                # By default, axes with an invisible patch forward events
+                # while axes with a visible patch capture events.
+                if a.get_capture_navigation_events() == "auto":
+                    if a.patch.get_visible():
+                        axes_capture[True].setdefault(
+                            a.get_zorder(), []).append(a)
+                    else:
+                        axes_capture[False].setdefault(
+                            a.get_zorder(), []).append(a)
+                else:
+                    axes_capture[a.get_capture_navigation_events()].setdefault(
+                        a.get_zorder(), []).append(a)
+
+        if not any(axes_capture.values()):
             return
+
         if self._nav_stack() is None:
-            self.push_current()  # set the home button to this view
+            self.push_current()   # Set the home button to this view.
+
+        # Handle the topmost axes that captures navigation (if there is one).
+        if axes_capture[True]:
+            zorder = max(axes_capture[True])
+            ax = axes_capture[True][zorder][-1]
+            ax.start_pan(event.x, event.y, event.button)
```

Comment:
```suggestion
            ax.start_zoom(event.x, event.y, event.button)
```

There seems to be a lot of code duplication between the pan and zoom, would it make sense to break some of it out into a private helper?